include/swift/AST/ASTMangler.h

```diff
@@ -384,7 +384,22 @@ class ASTMangler : public Mangler {
   static Optional<SpecialContext>
   getSpecialManglingContext(const ValueDecl *decl, bool useObjCProtocolNames);
 
-  static const clang::NamedDecl *
+  struct ClangDeclForManglingContainer {
```

Comment:
Open to suggestions for a better name.

Reply:
Is this meant to wrap the name mangling functionality or is it meant to wrap the NameDecl as a bool?

Reply:
Its just a mechanism for cleaning up the temporary name change we make to make mangling consistent. See one of my other comments.